### PR TITLE
Auth0 quickstart for Neon Authorize

### PR DESCRIPTION
**Pull request by bgrenon**
https://neon-next-git-bgrenon-autho-quickstart-neondatabase.vercel.app/docs/guides/neon-authorize-auth0

### CODE REVIEW

File: content/docs/guides/neon-authorize-auth0.md [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+<DocsList title="Related docs" theme="docs">
+  <a href="/docs/guides/neon-authorize-tutorial">Neon Authorize Tutorial</a>
+</DocsList>
+</InfoBlock>
+
+Use Auth0 with Neon Authorize to add secure, database-level authorization to your application. This guide assumes you already have an application using Auth0 for user authentication. It shows you how to integrate Auth0 with Neon Authorize, then provides sample Row-level Security (RLS) policies to help you model your own application schema.
+
+## How it works
+
+Auth0 handles user authentication by generating JSON Web Tokens (JWTs), which are securely passed to Neon Authorize. Neon Authorize validates these tokens and uses the embedded user identity metadata to enforce the [Row-Level Security](https://neon.tech/postgresql/postgresql-administration/postgresql-row-level-security) policies that you define directly in Postgres, securing database queries based on that user identity.
```
Comment (davidgomes) [inline content/docs/guides/neon-authorize-auth0.md:21]: would be nice to add: "This is all powered by the open source pg_session_jwt Postgres extension" and have a github link there

File: content/docs/guides/neon-authorize-auth0.md [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+    id: bigint('id', { mode: 'bigint' }).primaryKey().generatedByDefaultAsIdentity(),
+    userId: text('user_id')
+      .notNull()
+      .default(sql`(auth.user_id())`),
+    task: text('task').notNull(),
+    isComplete: boolean('is_complete').notNull().default(false),
+    insertedAt: timestamp('inserted_at', { withTimezone: true }).defaultNow().notNull(),
+  },
+  (t) => ({
+    p1: pgPolicy('create todos', {
```
Comment (davidgomes) [inline content/docs/guides/neon-authorize-auth0.md:164]: we should really use crudPolicy here - it's so much simpler

**Review by davidgomes — APPROVED** [review]

File: content/docs/guides/neon-authorize-auth0.md (hunk 2 above)
Comment (bgrenon) [reply content/docs/guides/neon-authorize-auth0.md:164]: these are sticking mostly to what's in the UI - but we dont' have to do that.

Comment (bgrenon) [reply content/docs/guides/neon-authorize-auth0.md:164]: will update the guides for that next week

File: content/docs/guides/neon-authorize-auth0.md [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+## How it works
+
+Auth0 handles user authentication by generating JSON Web Tokens (JWTs), which are securely passed to Neon Authorize. Neon Authorize validates these tokens and uses the embedded user identity metadata to enforce the [Row-Level Security](https://neon.tech/postgresql/postgresql-administration/postgresql-row-level-security) policies that you define directly in Postgres, securing database queries based on that user identity. This authorization flow is made possible using the Postgres extension [pg_session_jwt](https://github.com/neondatabase/pg_session_jwt), which you'll install as part of this guide.
+
+## Prerequisites
+
+To follow along with this guide, you will need:
+
+- A Neon account. Sign up at [Neon](https://neon.tech) if you don't have one.
+- A [Auth0](https://auth0.com/) account with an existing application (e.g., a **todos** app) that uses Auth0 for user authentication. If you don't have an app, check our [demo](https://github.com/neondatabase-labs/auth0-nextjs-neon-authorize) for similar schema and policies in action.
```
Comment (danieltprice) [inline content/docs/guides/neon-authorize-auth0.md:28]:
```suggestion
- An [Auth0](https://auth0.com/) account with an existing application (e.g., a **todos** app) that uses Auth0 for user authentication. If you don't have an app, check our [demo](https://github.com/neondatabase-labs/auth0-nextjs-neon-authorize) for similar schema and policies in action.
```

File: content/docs/guides/neon-authorize-auth0.md [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+GRANT USAGE ON SCHEMA public TO authenticated;
+GRANT USAGE ON SCHEMA public TO anonymous;
+```
+
+- **Authenticated role**: This role is intended for users who are logged in. Your application should send the authorization token when connecting using this role.
+- **Anonymous role**: This role is intended for users who are not logged in. It should allow limited access, such as reading public content (e.g., blog posts) without authentication.
+
+### 5. Install the Neon Serverless Driver
+
+Neon’s Serverless Driver manages the connection between your application and the Neon Postgres database. For Neon Authorize, you must use HTTP. While it is technically possible to access the HTTP API without using our driver, we recommend using the driver for best performance. The driver also supports WebSockets and TCP connections, so make sure you use the HTTP method when working with Neon Authorize.
```
Comment (danieltprice) [inline content/docs/guides/neon-authorize-auth0.md:105]:
```suggestion
Neon’s Serverless Driver manages the connection between your application and the Neon Postgres database. For Neon Authorize, you must use HTTP. While it is technically possible to access Neon's HTTP API without using our driver, we recommend using the driver for best performance. The driver supports connecting over both WebSockets and HTTP, so make sure you use the [HTTP connection method](/docs/serverless/serverless-driver#use-the-driver-over-http) when working with Neon Authorize.
```

**Review by danieltprice — APPROVED** [review]

File: content/docs/guides/neon-authorize-auth0.md (hunk 1 above)
Comment (bgrenon) [reply content/docs/guides/neon-authorize-auth0.md]: done

**Pull request merged**